### PR TITLE
Enable Storybook docs and MDX for long-form documentation

### PR DESCRIPTION
**Pull request by dhruvkb**
**Description**
This PR enables MDX and the Docs addon for Storybook. This will allow us to write long-form documentation adjacent to the stories for the components.

**Type of PR** 
This PR is a feature.

**Technicalities**
This PR will be followed by porting our stories from `component.stories.js` to `component.stories.mdx` and a corresponding change in the `.storybook/main.js` file to allow stories in both formats.

**Tests**
- See the Netlify deployment for this PR.

**Checklist:**
 
- [x] My pull request has a descriptive title (not a vague title like "Update `index.md`").
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Legal**
By submitting this PR, I agree to abide by the terms of the Developer 
Certificate of Origin.

<details>
<summary>Developer Certificate of Origin</summary>

Developer Certificate of Origin<br>
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.<br>
1 Letterman Drive<br>
Suite D4700<br>
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
</details>

### CODE REVIEW

**Review by hugosolar — APPROVED** [review]
I don't have any concerns about adding docs addon, I think it's necessary to have long-form documentation. But I think @brenoferreira had some concerns about the react dependency

**Comment by dhruvkb** [comment]
@hugosolar and @brenoferreira I had no choice as React is required by Storybook because MDX uses JSX for code snippets. It is a dev-dependency so it shouldn't affect the end-user. The final compiled library will have no React artifacts in the `dist/` directory and if we configure `package.json` using the `files` key, we can have strict control over what goes into the package when it's published to `npm`.

File: package.json [hunk 1]
```diff
@@ -46,5 +47,9 @@
   ],
   "license": "MIT",
   "repository": "github:creativecommons/vocabulary",
-  "main": "index.js"
+  "main": "index.js",
+  "dependencies": {
```
Comment (brenoferreira) [inline package.json]: Can we move this to `dev-dependencies`?

File: package.json [hunk 2]
```diff
@@ -46,5 +47,9 @@
...
+    "react": "^16.12.0",
```
Comment (brenoferreira) [inline package.json]: The react deps. should also be moved to `devDependencies`

**Review by brenoferreira — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (dhruvkb) [reply package.json]: @brenoferreira I addressed this, can you re-review?

**Review by brenoferreira — APPROVED** [review]

**Pull request merged**